adapters/repos/db/lsmkv/store.go

```diff
@@ -428,37 +428,64 @@ func (s *Store) ReplaceBuckets(ctx context.Context, bucketName, replacementBucke
 	}
 
 	s.bucketAccessLock.Lock()
-	defer s.bucketAccessLock.Unlock()
+	defer func() {
+		s.bucketAccessLock.Unlock()
+	}()
 
 	bucket := s.bucketsByName[bucketName]
 	if bucket == nil {
 		return fmt.Errorf("bucket '%s' not found", bucketName)
 	}
+
 	replacementBucket := s.bucketsByName[replacementBucketName]
 	if replacementBucket == nil {
 		return fmt.Errorf("replacement bucket '%s' not found", replacementBucketName)
 	}
 	s.bucketsByName[bucketName] = replacementBucket
 	delete(s.bucketsByName, replacementBucketName)
 
+	replacementBucket.disk.maintenanceLock.Lock()
+
 	currBucketDir := bucket.dir
 	newBucketDir := bucket.dir + "___del"
 	currReplacementBucketDir := replacementBucket.dir
 	newReplacementBucketDir := currBucketDir
 
+	if err := bucket.Shutdown(ctx); err != nil {
+
+		replacementBucket.disk.maintenanceLock.Lock()
```

Comment:
why are we locking here one more time? when we acquired this lock in line 447? shouldn't we Unlock here instead?